### PR TITLE
Add in ExpandPersistentVolumes as part of feature-gates on controller

### PR DESCRIPTION
**Pull request by kiich**
Compliments https://github.com/kubernetes-incubator/kube-aws/pull/1243 by adding `ExpandPersistentVolumes` as part of `--feature-gates` option when `PersistentVolumeClaimResize` is enabled via admission control.

Also noticed fix for https://github.com/kubernetes-incubator/kube-aws/issues/1255 so rolled the fix into `kubelet` part.

Ideally, would like to proceed further by supporting same method as workers by supporting following for `controller` block:
```
featureGates:
        ExpandPersistentVolumes: true
```

### CODE REVIEW

**Comment by mumoshu** [comment]
@kiich Thanks! In the future, we should make it more general so that we won't need to add available feature gates one by one..

**Comment by mumoshu** [comment]
@kiich Would you mind rebasing?

**Comment by kiich** [comment]
@mumoshu I was actually looking into how the `featureGates` works so would be up for PR-ing similar/same logic for the rest. Will carry on after upstreaming PR for https://github.com/kubernetes-incubator/kube-aws/issues/1204 which I'm nearly done with.

**Comment by kiich** [comment]
@mumoshu re: rebasing - i just fixed the conflict in github editor - was this enough? (sorry for obvious questions)

File: core/controlplane/config/templates/cloud-config-controller [hunk 1]
```diff
@@ -348,9 +348,9 @@ coreos:
         {{ else }}--cluster-dns={{.DNSServiceIP}} \
         {{ end }}--cluster-domain=cluster.local \
         --cloud-provider=aws \
-        {{if .Experimental.Admission.Priority.Enabled}}
-        --feature-gates=PodPriority=true \
-        {{end}}\
+        {{if or (.Experimental.Admission.Priority.Enabled) (.Experimental.Admission.PersistentVolumeClaimResize.Enabled) -}}
+        --feature-gates=PodPriority={{.Experimental.Admission.Priority.Enabled}},ExpandPersistentVolumes={{.Experimental.Admission.PersistentVolumeClaimResize.Enabled}} \
+        {{end -}}\
```
Comment (mumoshu) [inline core/controlplane/config/templates/cloud-config-controller]: Good try! But I guess the trailing slash still breaks the commmand. Let'ts fix it if this turns out to remain problematic.

**Comment by mumoshu** [comment]
@kiich LGTM. Thanks a ton for the fix!

Regarding #1255, I have a concern noted in https://github.com/kubernetes-incubator/kube-aws/pull/1293#discussion_r187495186. I appreciate it if you could confirm that.

File: core/controlplane/config/templates/cloud-config-controller (hunk 1 above)
Comment (kiich) [reply core/controlplane/config/templates/cloud-config-controller]: Hi @mumoshu 
I thought so as well but it seems to be working/not showing the same problem as #1255 
Whenever i toggle either of `Priority` or `PersistentVolumeClaimResize`, the resulting kubelet service seems to pass ok.
e.g. When I enable `Priority`, the following is shown for `kubelet.service` and Controllers seems to come up.

```
--cloud-provider=aws \
--feature-gates=PodPriority=true,ExpandPersistentVolumes=false \
\
$KUBELET_OPTS
Restart=always
RestartSec=10```

As you can see, no spaces/lines that was causing the issue in #1255 as far as I can tell.

**Comment by kiich** [comment]
Thanks @mumoshu - I can confirm the blank lines seen in #1255 does not seem to occur and controllers coming up ok as far as I can tell.

File: core/controlplane/config/templates/cluster.yaml [hunk 2]
```diff
@@ -1319,6 +1322,8 @@ experimental:
       enabled: false
     OwnerReferencesPermissionEnforcement:
       enabled: false
+    # persistentVolumeClaimResize enables PersistentVolumeClaim in the API server, scheduler and kubelet. you need to manually add the
+    # featureGate ExpandPersistentVolumes:true in the worker
```
Comment (c-knowles) [inline core/controlplane/config/templates/cluster.yaml]: @kiich @mumoshu, do you know when enabling the feature gate on node pools is necessary as per the comment? Is it only when control plane pods are scheduled onto workers?

**Pull request merged**